crowbar_framework/spec/controllers/api/restart_management_controller_spec.rb

```diff
@@ -0,0 +1,209 @@
+    end
+  end
+
+  context "POST sync" do
+    it "enabling changes the barclamp disallow_restart status to true" do
+      # disable_restart should not exists prior to the call
+      expect(@testing_node["nova"][:disallow_restart]).to be(nil)
+
+      post "/api/restart_management/sync", { disallow_restart: true, barclamp: "nova" }, headers
+      expect(response).to have_http_status(:ok)
+      # now the flag should be on true
+      expect(@testing_node["nova"][:disallow_restart]).to be(true)
+    end
+
+    it "disabling changes the barclamp disallow_restart status to false" do
+      # disable_restart should not exists prior to the call
+      expect(@testing_node["nova"][:disallow_restart]).to be(nil)
+
+      post "/api/restart_management/sync", { disallow_restart: false, barclamp: "nova" }, headers
+      expect(response).to have_http_status(:ok)
+      # now the flag should be on false
+      expect(@testing_node["nova"][:disallow_restart]).to be(false)
+    end
+
+    it "alternate calls to enable/disable set the barclamp disallow restart status properly" do
+      # disable_restart should not exists prior to the call
+      expect(@testing_node["nova"][:disallow_restart]).to be(nil)
+
+      post "/api/restart_management/sync", { disallow_restart: true, barclamp: "nova" }, headers
+      expect(response).to have_http_status(:ok)
+      # now the flag should be on false
+      expect(@testing_node["nova"][:disallow_restart]).to be(true)
+
+      post "/api/restart_management/sync", { disallow_restart: false, barclamp: "nova" }, headers
+      expect(response).to have_http_status(:ok)
+      # now the flag should be on false
+      expect(@testing_node["nova"][:disallow_restart]).to be(false)
+
+      post "/api/restart_management/sync", { disallow_restart: true, barclamp: "nova" }, headers
+      expect(response).to have_http_status(:ok)
+      # now the flag should be on false
+      expect(@testing_node["nova"][:disallow_restart]).to be(true)
+    end
+
+    it "returns a 404 error if we pass a non-existant barclamp" do
+      post "/api/restart_management/sync", { disallow_restart: true, barclamp: "parrot" }, headers
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it "returns a 409 error if the proposal is not active" do
+      @catalog.update("parrot" => 50)
+      allow(Proposal).to receive(:where).with(barclamp: "parrot").and_return([])
+      post "/api/restart_management/sync", { disallow_restart: true, barclamp: "parrot" }, headers
+      expect(response).to have_http_status(:conflict)
+    end
+  end
+
+  context "GET restarts" do
+    it "lists the nodes and the services that require service restarts" do
+      @admin_node.set["nova"] = {}
+      @admin_node.set["neutron"] = {}
+      @testing_node.set["cinder"] = {}
+      time = Time.now.getutc.to_s
+      @admin_node.set["nova"]["requires_restart"] = {
+        "openstack-nova-api" => {
+          "pacemaker_service" => false,
```

Comment:
Maybe setting one of the `pacemaker_service` datas to `true` would cover that part as well? Also I think then 2 test vectors should be enough?

Reply:
Well, in here the outcome of the test is more to see that the output is as we expect, more than to check the proper values, especially as the `pacemaker_service` and `timestamp` values are set on the chef runs and we have no way of testing them, so those attributes could be empty for all we care. So only testing the output rather than the actual values or anything like that.